service/src/main/java/bio/terra/catalog/service/dataset/DatasetDao.java

```diff
@@ -0,0 +1,120 @@
+package bio.terra.catalog.service.dataset;
+
+import bio.terra.catalog.common.DaoKeyHolder;
+import bio.terra.catalog.common.StorageSystem;
+import bio.terra.catalog.service.dataset.exception.DatasetNotFoundException;
+import bio.terra.catalog.service.dataset.exception.InvalidDatasetException;
+import bio.terra.common.db.ReadTransaction;
+import bio.terra.common.db.WriteTransaction;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.List;
+import java.util.UUID;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class DatasetDao {
+
+  private final NamedParameterJdbcTemplate jdbcTemplate;
+  private static final String ID_FIELD = "id";
+  private static final String DATASET_ID_FIELD = "dataset_id";
```

Comment:
The more I see this, the more I think we should have a different name for this field. If I was writing a comment in the code and was talking about "dataset id", which ID would I be talking about? I think the "id" field on "dataset" would be the "dataset id". Maybe this should be "storage ID" (or "resource ID")?